### PR TITLE
OCPBUGS-24606: UPSTREAM: <carry>: Update management webhook pod admission logic

### PR DESCRIPTION
**Pull request by eggfoobar**
Updating the logic for pod admission to allow a pod creation with workload partitioning annotations to be run in a namespace that has no workload allow annoations.

The pod will be stripped of its workload annotations and treated as if it were normal, a warning annoation will be placed to note the behavior on the pod.

See more info https://issues.redhat.com/browse/CNF-10581

### CODE REVIEW

File: openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go [hunk 1]
```diff
@@ -217,6 +217,11 @@ func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attri
 		return err
 	}
 
+	if _, found := ns.Annotations[namespaceAllowedAnnotation]; !found && len(workloadType) > 0 {
+		pod.Annotations[workloadAdmissionWarning] = "skip pod CPUs requests modifications because namespace is not annotated for workloads"
```
Comment (MarSik) [inline openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go:221]: I think you should be more specific (keywords that allow googling the docs) in the message. I am not sure if the text clearly points at the workload partitioning feature. Think about people hearing about it for the first time.

**Review by MarSik — CHANGES_REQUESTED** [review]
I am ok with the change, I just think the message should be clearer and allow efficient searching for the relevant docs.

File: openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go [hunk 2]
```diff
@@ -449,10 +454,13 @@ func podHasBothCPULimitAndRequest(containers []coreapi.Container) bool {
 	return false
 }
 
+// doesNamespaceAllowWorkloadType will return false when a workload type does not match any present ones.
+// When a namespace contains no annotation for workloads we infer that to mean all workload types are allowed.
+// Mutation hook will strip all workload annotation from pods that contain workload annotations in that circumstance.
 func doesNamespaceAllowWorkloadType(annotations map[string]string, workloadType string) bool {
 	v, found := annotations[namespaceAllowedAnnotation]
```
Comment (Tal-or) [inline openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go:461]: The reverse of the condition is a bit confusing. I would at least place the comment right above the `return`

File: openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go (hunk 1 above)
Comment (eggfoobar) [reply openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go]: Thats a good point, updated language with a bit more detail to help with that

File: openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go (hunk 2 above)
Comment (eggfoobar) [reply openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go:461]: Good idea, moved it to the return logic to help with the explanation

**Review by MarSik — APPROVED** [review]
/lgtm

**Review by soltysh — APPROVED** [review]
/approve

**Comment by soltysh** [comment]
/override ci/prow/unit
the failure is a known one

**Pull request merged**